tests/std/tests/GH_004930_char_traits_user_specialization/test.cpp

```diff
@@ -280,6 +285,16 @@ CONSTEXPR20 bool test_gh_4930() {
 static_assert(test_gh_4930());
 #endif // _HAS_CXX20
 
+void test_gh_4956() {
+    basic_string<char, odd_char_traits<char>> s("QQPPQ", 5);
+
+    bitset<7> bs;
+    std::basic_stringstream<char, odd_char_traits<char>>(s) >> bs;

```

Comment:
This is constructing a temporary with parentheses. I'd prefer to split the construction and streaming to improve clarity.

Also, this can be a `basic_istringstream` - we aren't performing output here.

Also, `std::`.